hoomd/hpmc/tune/move_size.py

```diff
@@ -0,0 +1,333 @@
+from hoomd.custom import _InternalAction
+from hoomd.parameterdicts import ParameterDict
+from hoomd.typeconverter import OnlyFrom, OnlyType, OnlyIf, to_type_converter
+from hoomd.tune import _InternalCustomTuner
+from hoomd.tune.attr_tuner import (
+    _TuneDefinition, SolverStep, ScaleSolver, SecantSolver)
+from hoomd.hpmc.integrate import _HPMCIntegrator
+
+
+class _MoveSizeTuneDefinition(_TuneDefinition):
+    """Encapsulates getting the acceptance rate and getting/setting move size.
+
+    This class should only be used for the _InternalMoveSize class to tune HPMC
+    move sizes. For this class 'x' is the move size and 'y' is the acceptance
+    rate.
+    """
+    _attr_acceptance = {
+        'a': 'rotate_moves',
+        'd': 'translate_moves'
+        }
+
+    def __init__(self, attr, type, target, domain=None):
+        if attr not in self._attr_acceptance:
+            raise ValueError("Only {} are allowed as tunable "
+                             "attributes.".format(self._available_attrs))
+        self.attr = attr
+        self.type = type
+        self.integrator = None
+        self.previous_ratio = None
+        self.previous_total = None
+        super().__init__(target, domain)
+
+    def _get_y(self):
+        current_ratio = getattr(self.integrator,
+                                self._attr_acceptance[self.attr])
+        current_total = sum(current_ratio)
+        # We return the target when no moves are recorded since we don't want
+        # the move size to be updated. This is "hackish", but there is no right
+        # answer here since there is no concept of an acceptance_rate with no
+        # trial moves. We could error, but I think this make the class easier to
+        # use for users.
+        if current_total == 0:
+            return self._target
```

Comment:
This hack is fine. We could alternately return `None` and handle that special case in the caller.

Reply:
I am not sure if I want to force the `SolverStep` objects to deal with a `y` of `None`. I am not sure how many cases of this problem there would be. If it is common enough, it may be worth baking into the `SolverStep` infrastructure, though. We could also just error and always use examples that use a `phase` with the trigger for attribute tuning operations.

Reply:
It's up to you. The other example case we have in mind for this (tuning r_buff for performance) is another example where the first value read will be undefined.